x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmTests.java

```diff
@@ -752,7 +753,7 @@ private ThreadContext createThreadContext(final CharSequence jwt, final CharSequ
         if (sharedSecret != null) {
             requestThreadContext.putHeader(
                 JwtRealm.HEADER_CLIENT_AUTHENTICATION,
-                JwtRealmSettings.CLIENT_AUTHENTICATION_TYPE_SHARED_SECRET + " " + sharedSecret
+                JwtRealm.HEADER_SHARED_SECRET_AUTHENTICATION_SCHEME + " " + sharedSecret
```

Comment:
Before, a single constant was used for the config value and request header with value `SharedSecret`. Now they are different constants with different values.
- `setting 'xpack.security.authc.realms.jwt.jwt2.client_authentication.type', 'shared_secret'`
- `X-Client-Authentication: SharedSecret test-secret`

That seems OK with me.

Reply:
That's correct, because they're different concepts. The previously had the same value, but there's no reason for them to - an HTTP authorization scheme is distinct from the value of a setting in yml.

Reply:
LGTM